readthedocs/api/v2/views/footer_views.py

```diff
@@ -63,113 +64,149 @@ def get_version_compare_data(project, base_version=None):
+        using_theme = (theme == 'default')
+
+        project = self._get_project()
+        main_project = project.main_language_project or project
+        version = self._get_version()
+
+        page_slug = self.request.GET.get('page', '')
+        if page_slug and page_slug != 'index':
+            if main_project.documentation_type == 'sphinx_htmldir':
+                path = page_slug + '/'
+            else:
+                path = page_slug + '.html'
+        else:
+            path = ''
+
+        context = {
+            'project': project,
+            'version': version,
+            'path': path,
+            'downloads': version.get_downloads(pretty=True),
+            'current_version': version.verbose_name,
+            'versions': project.ordered_active_versions(
+                user=self.request.user,
+            ),
+            'main_project': main_project,
+            'translations': main_project.translations.all(),
+            'current_language': project.language,
+            'using_theme': using_theme,
+            'new_theme': new_theme,
+            'settings': settings,
+            'subproject': subproject,
+            'github_edit_url': version.get_github_url(
+                docroot,
+                page_slug,
+                source_suffix,
+                'edit',
+            ),
+            'github_view_url': version.get_github_url(
+                docroot,
+                page_slug,
+                source_suffix,
+                'view',
+            ),
+            'gitlab_edit_url': version.get_gitlab_url(
+                docroot,
+                page_slug,
+                source_suffix,
+                'edit',
+            ),
+            'gitlab_view_url': version.get_gitlab_url(
+                docroot,
+                page_slug,
+                source_suffix,
+                'view',
+            ),
+            'bitbucket_url': version.get_bitbucket_url(
+                docroot,
+                page_slug,
+                source_suffix,
+            ),
+            'theme': theme,
+        }
+        return context
+
+    def get(self, request, format=None):
+        project = self._get_project()
+        version = self._get_version()
+        version_compare_data = get_version_compare_data(
+            project,
+            version,
+        )
+
+        context = self._get_context()
+        html = template_loader.get_template('restapi/footer.html').render(
+            context,
+            request,
+        )
+
+        resp_data = {
+            'html': html,
+            'show_version_warning': project.show_version_warning,
+            'version_active': version.active,
+            'version_compare': version_compare_data,
+            'version_supported': version.supported,
+        }
+
+        # Allow folks to hook onto the footer response for various information
+        # collection, or to modify the resp_data.
+        footer_response.send(
```

Comment:
This isn't used in any other part of the code.
Inheritance makes extending more easy anyway.

Are we ok removing it?

Reply:
We use it in `ext` to capture theme data, which is likely why it was added. I think it's fine to keep for now, and likely will be useful for capturing analytics data too.